src/app/core/common-components/entity-form/form-field-label/form-field-label.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Input, OnChanges, SimpleChanges } from "@angular/core";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { EntityConstructor } from "../../../entity/model/entity";
+import { ColumnConfig } from "../../entity-subrecord/entity-subrecord/entity-subrecord-config";
+import { FormFieldConfig } from "../entity-form/FormConfig";
+import { EntityFormService } from "../entity-form.service";
+import { NgIf } from "@angular/common";
+
+/**
+ * Generic component to display the label of one form field of an entity
+ * without having to handle overwriting the field config with potentially missing schema field details.
+ */
+@Component({
+  selector: "app-form-field-label",
+  templateUrl: "./form-field-label.component.html",
+  standalone: true,
+  imports: [MatTooltipModule, NgIf],
+})
+export class FormFieldLabelComponent implements OnChanges {
+  /** field id or full config */
+  @Input() field: ColumnConfig;
+  /** full field config extended from schema (used internally and for template) */
+  _field: FormFieldConfig;
+
+  /** optional alternative to passing an entity, you can provide an entity type only */
+  @Input() entityType: EntityConstructor;
+
+  constructor(private entityFormService: EntityFormService) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.field || changes.entityType) {
+      this.updateField();
+    }
+  }
+
+  private updateField() {
+    if (!this.entityType) {
+      this._field = undefined;
+      return;
+    }
+
+    this._field = this.entityFormService.extendFormFieldConfig(
```

Comment:
Does this really have to be another component which extends the config itself? In case of the entity subrecord this now means that is created 3 times there (1x `createFormGroup`, 1x `extendFormFieldConfig`, 1x this component). And when clicking on the row it happens another 2 times (1x row details `createFormGroup` and 1x in form field component). Currently this is at least only 3 times in total.

Reply:
the refactoring overall removes us calling getEntityConstructor as a method - that in itself has way way way more performance implication than (once on changes) doing an object.assign.

The alternative (which we decided against initially) could be to do the extending already in config-service or extremely early, centrally - so that everyone has the full details automatically. (but I do want to get back to "actual work" from all this refactoring here also :blush: )